### PR TITLE
Update webhook plugin doc

### PR DESCRIPTION
**Pull request by hezyin**
Remaining TODOs:

1. Format JSON payload schema into tables
2. Rewrite the Jenkins example in the Deployments section (please refer to the GitHub example)
3. Rewrite the following paragraph
>First, we need to know that CircleCi has three entities: pipeline, workflow, and job, and the entity workflow is the entity task in DevLake. Second, we must get pipelines and task data from the build machine. In CircleCi, the data define in env, and we can get it by $CIRCLE_WORKFLOW_JOB_ID and so on. So we can write config to send task data in each workflow and send the close pipeline request in the last workflow.

### CODE REVIEW

**Comment by likyh** [comment]
I have tried to do it. But it looks like the examples does not clarify `pipeline` and `task` in DevLake directly.

File: docs/Plugins/webhook.md [hunk 1]
```diff
@@ -42,35 +34,37 @@ If you want to collect issue or incident data from your system, you can use the
...
-creator_id(Optional): the user id of the creator
-creator_name(Optional): the user name of the creator, it will just be used to display
-assignee_id(Optional)
-assignee_name(Optional)
-severity(Optional)
-component(Optional): which component is this issue in.
-```
+needs to be called when an issue or incident is created. The body  should be a JSON and include columns as follows:
+
+|          Keyname          | Require | Notes                                                        |
```
Comment (keon94) [inline docs/Plugins/webhook.md]: I think it's better if the column name is "Required", and the values are Yes/No

File: docs/Plugins/webhook.md [hunk 2]
```diff
@@ -134,19 +140,21 @@ This hook should be added when the task starts and finishes. So in Example 3, we
...
-status: Must be one of IN_PROGRESS DONE
-type: one of TEST LINT BUILD DEPLOYMENT
-environment: which type of machine did this task run in. one of PRODUCTION STAGING TESTING
-started_date: date, Format should be 2020-01-01T12:00:00+00:00
-finished_date(Optional): date, Format should be 2020-01-01T12:00:00+00:00
-repo_id: build for which repo/project. It can be a unique string that you can distinguish
-branch(Optional)
-commit_sha(Optional)
-```
+| Keyname       | Require | Notes                                                        |
```
Comment (keon94) [inline docs/Plugins/webhook.md]: Same as above. "Required": Yes/No

File: docs/UserManuals/ConfigUI/webhook.md [hunk 3]
```diff
@@ -4,16 +4,16 @@ sidebar_position: 7
...
-Name your connection.
+#### Webhook name
 We recommand that you give your webhook connection a unique name so that you can identify and manage where you have used it later.
 
-### Use Webhook
-After cicking on "Generate POST URL", you will find four webhook URLs. Copy the ones that suits your usage into your CI or issue tracking systems. You can always come back to the webhook page to copy the URLs later on.
+### Use Webhooks
+After clicking on "Generate POST URL", you will find four webhook URLs. Copy the ones that suits your usage into your CI or issue tracking systems. You can always come back to the webhook page to copy the URLs later on.
 
 ![image](https://user-images.githubusercontent.com/3294100/191400110-327c153f-b236-47e3-88cc-85bf8fcae310.png)
```
Comment (keon94) [inline docs/UserManuals/ConfigUI/webhook.md:18]: Not a doc issue, but rather the wording of the [message](https://user-images.githubusercontent.com/3294100/191400110-327c153f-b236-47e3-88cc-85bf8fcae310.png) displayed in the UI:
"Copy the following URLs to your issue tracking tool for Incidents and CI tool for Deployments by making a POST to DevLake."
It sounds to me that we're asking the user to make a POST to Devlake, whereas we're wanting to say the CI/Tracking tool will need to make the POST call, right? I think the wording can be improved there. Thoughts @hezyin @e2corporation ?

File: docs/Plugins/webhook.md [hunk 4]
```diff
@@ -134,19 +140,21 @@ This hook should be added when the task starts and finishes. So in Example 3, we
...
+| status        | Require | one of `IN_PROGRESS` `DONE`                                  |
+| type          | Require | one of `TEST` `LINT` `BUILD` `DEPLOYMENT`                    |
+| environment   | Require | which type of machine did this task run in. one of `PRODUCTION` `STAGING` `TESTING` |
+| started_date  | Require | date, Format should be 2020-01-01T12:00:00+00:00             |
+| finished_date |         | date, Format should be 2020-01-01T12:00:00+00:00             |
+| repo_id       | Require | build for which repo/project. It can be a **unique** string that you can distinguish |
+| branch        |         |                                                              |
+| commit_sha    |         |                                                              |
+
+More information about these column at [DomainLayerCicdTask](https://devlake.apache.org/docs/DataModels/DevLakeDomainLayerSchema#cicd_tasks).
```
Comment (keon94) [inline docs/Plugins/webhook.md]: fix: More information about these **_columns_** at

File: docs/Plugins/webhook.md [hunk 5]
```diff
@@ -100,27 +94,39 @@ Read more in Swagger: http://localhost:4000/api/swagger/index.html#/plugins%2Fwe
...
+- 12 entry in cicd_pipelines table, corresponds to the Jenkins builds
 
 ![image](https://user-images.githubusercontent.com/3294100/191319924-f05c4790-d368-4fe4-8c07-dea43e1dd2f3.png)
+(Image from the Internet)
 
-(Example 2. Image from the Internet)
+#### Example 3
 
-In Example 2, we can find 12 Jenkins pipelines.
+Jenkins build may contain some stage. This screenshot shows 5 Jenkins builds, which consists of 1~4 stages including `Build the sudo images for installation`, `Basic Test: running from repo with preinstalled libs`, and etc. When DevLake normalizes data from Jenkins, it creates the following records in the domain layer:
```
Comment (keon94) [inline docs/Plugins/webhook.md]: wording/semantics fix:

Jenkins builds may contain stages. This screenshot shows 5 Jenkins builds, which consist of 1~4 stages, including `Build the sudo images for installation`, `Basic Test: running from repo with preinstalled libs`, etc. When DevLake gets and normalizes data from Jenkins, it creates the following records in the domain layer:

File: docs/Plugins/webhook.md [hunk 6]
```diff
@@ -172,10 +180,10 @@ curl http://127.0.0.1:4000/api/plugins/webhook/1/cicd_pipeline/A123/finish -X 'P
 
 Read more in Swagger: http://localhost:4000/api/swagger/index.html#/plugins%2Fwebhook/post_plugins_webhook__connectionId_issues. 
 
-### Sample Config in CircleCi
+### Sample Config in CircleCI
 
-First, we need to know that CircleCi has three entities: pipeline, workflow, and job, and the entity workflow is the entity task in DevLake. 
-Second, we must get pipelines and task data from the build machine. In CircleCi, the data define in env, and we can get it by $CIRCLE_WORKFLOW_JOB_ID and so on. So we can write config to send task data in each workflow and send the close pipeline request in the last workflow.
+CircleCI pipelines are the highest-level unit of work. Pipelines include your workflows, which coordinate your jobs. The following demo regard CircleCI workflow as the entity task in DevLake.
```
Comment (keon94) [inline docs/Plugins/webhook.md]: Wording/semantics fix:

CircleCI pipelines are the highest-level unit of work. Pipelines include your workflows, which coordinate your jobs. The following demo is regarding a CircleCI workflow running as an entity task in DevLake.

File: docs/Plugins/webhook.md [hunk 7]
```diff
@@ -172,10 +180,10 @@ curl http://127.0.0.1:4000/api/plugins/webhook/1/cicd_pipeline/A123/finish -X 'P
...
+In CircleCI, the data defined in env describe build machine, pipelines and tasks , sucs as `$CIRCLE_WORKFLOW_JOB_ID`, etc. So Write config to send task data read from env in each workflow and send the close pipeline request in the last workflow.
```
Comment (keon94) [inline docs/Plugins/webhook.md:65]: Wording/semantics fix:

In CircleCI, the data defined in _env_ describe the build machine, pipelines and tasks (e.g. `$CIRCLE_WORKFLOW_JOB_ID`), etc. You will need to add config to send task data, read from the _env_, for each workflow, and to send the _close pipeline_ request in the last workflow.

File: docs/Plugins/webhook.md [hunk 8]
```diff
@@ -220,7 +228,7 @@ workflows:
 
 
 
-Actually, we finish the webhook in prev step. If we want to do more, the config in CircleCi is as fellow. It will call webhook before tasks start and after tasks fail.
+Actually, the incoming webhook works in prev step. If you want to do more, the config in CircleCI is as fellow. It will call webhook before tasks start and after tasks fail.
```
Comment (keon94) [inline docs/Plugins/webhook.md]: I'm not understanding the first sentence: (Actually, the incoming webhook works in prev step). The original version made more sense.
As for the rest, replace with:

If you want to customize more, a sample CircleCI config is provided below. This sample calls the webhook before the tasks start and after tasks fail.

File: docs/Plugins/webhook.md [hunk 9]
```diff
@@ -100,27 +94,39 @@ Read more in Swagger: http://localhost:4000/api/swagger/index.html#/plugins%2Fwe
 
 ## Deployments
 
-Adding pipeline webhook in your deploy shells will help you collect data into DevLake.
+If your CI/CD tool isn't already supported by DevLake, you can insert curl commands in your CI/CD script to post deployment data to DevLake.
+
+DevLake models CI/CD domain with two entities: Pipelines and Tasks. A pipeline has many tasks.
```
Comment (keon94) [inline docs/Plugins/webhook.md]: Semantics/Wording fix:

DevLake CI/CD domain-models contain two entities: Pipelines and Tasks. A pipeline can have one or more tasks.

File: docs/Plugins/webhook.md [hunk 10]
```diff
@@ -4,34 +4,26 @@ description: >
...
-(Image from the Internet)
-
-While using webhooks in DevLake, your systems play the roles of system A as shown in the above image, and Apache DevLake as system B. You can use webhooks to:
-
-* record bugs or incidents when DevLake cannot collect data from issue applications directly.
-* detect code deployments by using custom deploy-time measurement 
+When you create an Incoming Webhook within DevLake, DevLake generates a unique URL. You can post JSON payloads to this URL to push data to DevLake.
 
-DevLake does not limit the number of webhook requests. **But be aware** that your Database or other servers may not handle unlimited requests.
+As of v0.14.0, users can push incidents and deployments data required by DORA metrics to DevLake via Incoming Webhook.
```
Comment (keon94) [inline docs/Plugins/webhook.md]: fix: via Incoming Webhooks

File: docs/Plugins/webhook.md [hunk 11]
```diff
@@ -4,34 +4,26 @@ description: >
...
 
-## Configuration
-Configuring webhooks via the Config UI.
+## Creating webhooks in DevLake
 
-First, you can create a webhook connection from the Integration page. 
+### Add a new webhook
+To add a new webhook, go to "Data Connections" page in config-ui and select "Issue/Deployment Incoming/Webhook".
```
Comment (keon94) [inline docs/Plugins/webhook.md]: fix: go to the "Data Connections"

File: docs/Plugins/webhook.md [hunk 12]
```diff
@@ -4,34 +4,26 @@ description: >
...
 ![image](https://user-images.githubusercontent.com/3294100/191309840-460fbc9c-15a1-4b12-a510-9ed5ccd8f2b0.png)
 
 We recommand that you give your webhook connection a unique name so that you can identify and manage where you have used it later.
 
-After cicking on "Generate POST URL", you will find four webhook URLs. Copy the ones that suits your usage into your CI or issue tracking systems. You can always come back to the webhook page to copy the URLs later on.
+After clicking on "Generate POST URL", you will find four webhook URLs. Copy the ones that suits your usage into your CI or issue tracking systems. You can always come back to the webhook page to copy the URLs later on.
```
Comment (keon94) [inline docs/Plugins/webhook.md]: fix: After clicking on the "Generate POST URL" button, you will find four webhook URLs. Copy the ones that suit your usage into your CI or issue tracking systems.

File: docs/Plugins/webhook.md [hunk 13]
```diff
@@ -100,27 +94,39 @@ Read more in Swagger: http://localhost:4000/api/swagger/index.html#/plugins%2Fwe
...
 
 ![image](https://user-images.githubusercontent.com/3294100/191319143-ea5e9546-1c6d-4b2a-abba-95375cfdcec3.png)
 
-(Example 1)
+#### Example 2
+
+Then look at Jenkins build history, the screenshot below shows 12 Jenkins builds, and no Jenkins stages appeared. When DevLake normalizes data from Jenkins build history, it creates the following records in the domain layer:
 
-For example, we can find 6 tasks like `golangci-lint` and `unit-test` in one GitHub pipeline.
+- 12 entry in cicd_pipelines table, corresponds to the Jenkins builds
```
Comment (keon94) [inline docs/Plugins/webhook.md]: 12 entries in cicd_pipelines table, corresponding to the Jenkins builds

File: docs/Plugins/webhook.md [hunk 14]
```diff
@@ -100,27 +94,39 @@ Read more in Swagger: http://localhost:4000/api/swagger/index.html#/plugins%2Fwe
...
 
-Adding pipeline webhook in your deploy shells will help you collect data into DevLake.
+If your CI/CD tool isn't already supported by DevLake, you can insert curl commands in your CI/CD script to post deployment data to DevLake.
+
+DevLake models CI/CD domain with two entities: Pipelines and Tasks. A pipeline has many tasks.
+
+#### Example 1
 
-First, let us know there are two entities: Tasks and Pipelines. A pipeline means one build or deployment, and a pipeline may have more than one task.
+Take GitHub Actions as an example, the screenshot below shows a GitHub Actions workflow, which consists of 6 jobs including `golangci-lint`, `unit-test`, and etc. When DevLake normalizes data from GitHub Actions, it creates the following records in the domain layer:
```
Comment (keon94) [inline docs/Plugins/webhook.md]: Take GitHub Actions as an example:

File: docs/Plugins/webhook.md [hunk 15]
```diff
@@ -100,27 +94,39 @@ Read more in Swagger: http://localhost:4000/api/swagger/index.html#/plugins%2Fwe
...
+
+- 1 entry in cicd_pipelines table, corresponds to the GitHub workflow
+- 6 entries in cicd_tasks table, one for each job in the GitHub workflow
 
 ![image](https://user-images.githubusercontent.com/3294100/191319143-ea5e9546-1c6d-4b2a-abba-95375cfdcec3.png)
 
-(Example 1)
+#### Example 2
+
+Then look at Jenkins build history, the screenshot below shows 12 Jenkins builds, and no Jenkins stages appeared. When DevLake normalizes data from Jenkins build history, it creates the following records in the domain layer:
```
Comment (keon94) [inline docs/Plugins/webhook.md]: Now look at the Jenkins build history, as another example: the screenshot below shows 12 Jenkins builds, with no Jenkins stages:

**Comment by likyh** [comment]
I have copied some review docs.

**Comment by likyh** [comment]
I have deleted the old webhook doc and added new one.

**Review by Startrekzky — APPROVED** [review]
LGTM

**Pull request merged**